File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt = 1;
        int el = nums[0];
        for(int i = 1; i<nums.size(); i++){
            if(nums[i]==el){
                cnt++;
            }else if(cnt==0){
                el = nums[i];
                cnt++;
            }else{
                cnt--;
            }
        }
        return el;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int el = 0;
        int cnt = 0;
        for(int num: nums){
            if(cnt==0){
                el = num;
                cnt++;
            }
            else if(el == num){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        return el;
    }
}